website/docs/r/aws_auth_backend_cert.md

```diff
@@ -0,0 +1,59 @@
+---
+layout: "vault"
+page_title: "Vault: vault_aws_auth_backend_cert resource"
+sidebar_current: "docs-vault-aws-auth-backend-cert"
+description: |-
+  Manages a certificate for an AWS Auth Backend in Vault.
+---
+
+# vault\_aws\_auth\_backend\_cert
+
+Manages a certificate to be used with an AWS Auth Backend in Vault.
+
+This resource sets the AWS public key and the type of document that can be
+verified against the key that Vault can then use to verify the instance
+identity documents making auth requests.
+
+For more information, see the [Vault
+docs](https://www.vaultproject.io/api/auth/aws/index.html#configure-client).
+
+~> **Important** All data provided in the resource configuration will be
+written in cleartext to state and plan files generated by Terraform, and will
+appear in the console output when Terraform runs. Protect these artifacts
+accordingly. See [the main provider documentation](../index.html) for more
+details.
+
+## Example Usage
+
+```hcl
+resource "vault_auth_backend" "aws" {
+  type = "aws"
+}
+
+resource "vault_aws_auth_backend_cert" "cert" {
+  backend = "${vault_auth_backend.aws.path}"
+  cert_name = "my-cert"
+  aws_public_cert = "LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUM3VENDQXEwQ0NRQ1d1a2paNVY0YVp6QUpCZ2NxaGtqT09BUURNRnd4Q3pBSkJnTlZCQVlUQWxWVE1Sa3cKRndZRFZRUUlFeEJYWVhOb2FXNW5kRzl1SUZOMFlYUmxNUkF3RGdZRFZRUUhFd2RUWldGMGRHeGxNU0F3SGdZRApWUVFLRXhkQmJXRjZiMjRnVjJWaUlGTmxjblpwWTJWeklFeE1RekFlRncweE1qQXhNRFV4TWpVMk1USmFGdzB6Ck9EQXhNRFV4TWpVMk1USmFNRnd4Q3pBSkJnTlZCQVlUQWxWVE1Sa3dGd1lEVlFRSUV4QlhZWE5vYVc1bmRHOXUKSUZOMFlYUmxNUkF3RGdZRFZRUUhFd2RUWldGMGRHeGxNU0F3SGdZRFZRUUtFeGRCYldGNmIyNGdWMlZpSUZObApjblpwWTJWeklFeE1RekNDQWJjd2dnRXNCZ2NxaGtqT09BUUJNSUlCSHdLQmdRQ2prdmNTMmJiMVZRNHl0LzVlCmloNU9PNmtLL24xTHpsbHI3RDhad3RRUDhmT0VwcDVFMm5nK0Q2VWQxWjFnWWlwcjU4S2ozbnNzU05wSTZiWDMKVnlJUXpLN3dMY2xuZC9Zb3pxTk5tZ0l5WmVjTjdFZ2xLOUlUSEpMUCt4OEZ0VXB0M1FieVlYSmRtVk1lZ042UApodmlZdDVKSC9uWWw0aGgzUGExSEpkc2tnUUlWQUxWSjNFUjExK0tvNHRQNm53dkh3aDYrRVJZUkFvR0JBSTFqCmsrdGtxTVZIdUFGY3ZBR0tvY1Rnc2pKZW02LzVxb216SnVLRG1iSk51OVF4dzNyQW90WGF1OFFlK01CY0psL1UKaGh5MUtIVnBDR2w5ZnVlUTJzNklMMENhTy9idXljVTFDaVlRazQwS05IQ2NIZk5pWmJkbHgxRTlycFVwN2JuRgpsUmEydjFudE1YM2NhUlZEZGJ0UEVXbWR4U0NZc1lGRGs0bVpyT0xCQTRHRUFBS0JnRWJtZXZlNWY4TElFL0dmCk1ObVA5Q001ZW92UU9HeDVobzhXcUQrYVRlYnMrazJ0bjkyQkJQcWVacXBXUmE1UC8ranJkS21sMXF4NGxsSFcKTVhyczNJZ0liNitoVUlCK1M4ZHo4L21tTzBicHI3NlJvWlZDWFlhYjJDWmVkRnV0N3FjM1dVSDkrRVVBSDVtdwp2U2VEQ09VTVlRUjdSOUxJTll3b3VISXppcVFZTUFrR0J5cUdTTTQ0QkFNREx3QXdMQUlVV1hCbGs0MHhUd1N3CjdIWDMyTXhYWXJ1c2U5QUNGQk5HbWRYMlpCclZOR3JOOU4yZjZST2swazlLCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K"
```

Comment:
I'm a little ambivalent about putting this here. While it's nice to have ready-to-run examples, I expect this really long line will cause some ill-effects on the website (even if not in the _current_ design, it's a potential area of conflict for future redesigns) and it's pretty opaque for the user anyway.

Could we instead make this be an example of reading from a separate file, with a comment explaining how to generate that file? Something like this:

```hcl
  # Generate this file using ....
  aws_public_cert = "${file("${path.module}/aws_auth.crt")}"
```

(Hopefully there's some docs either in AWS or in Vault that we can link to, rather than specifically suggesting a command here in our own docs.)

---

Also general doc-related observation while I'm here:

I think it's best if we try to keep the doc examples consistent with what `terraform fmt` would generate, which includes aligning all the equals signs when single-line attributes appear one after another:

```hcl
  backend         = "..."
  cert_name       = "..."
  aws_public_cert = "..."
  type            = "..."
```

Reply:
I like your file approach better, and the `terraform fmt` comment is a good reminder, thanks. I've been applying it to other PRs as I merged them. I'll make sure everything you reviewed today gets that treatment.